enhancements/baremetal/debuggability-of-baremetal-ipi-deployment.md

```diff
@@ -0,0 +1,246 @@
+is that Ironic is successful in cleaning, inspecting, and deploying a
+host. After Ironic lays down an image on disk and reboots, Ironic marks
+the host ‘active’.  However, when the host boots again it’s possible that
+there’s a catastrophic problem such as a kernel panic or fail to
+configure with ignition. From Ironic's perspective, it's done it's duty,
+and is unaware the host failed to come up. The feedback to the user is
+only that there was a timeout.
+
+#### Operator Deployment
+
+Operator deployment failures are rarely platform-specific, although
+there is one case that should be addressed. When worker deployment
+fails, possibly due to provisioning issues like those described above, a
+variety of operators may report failures such as ingress, console, and
+others that cannot run on the control plane.
+
+When this happens, the installer times out, reports to the user a large
+number of operators failed to roll out, and no useful context about what
+to do or why the operators failed.
+
+#### User Stories
+
+##### Show more information from terraform
+
+  - As a user, I want terraform to report last_error and status from
+    ironic in case of deployment failure.
+
+  - As a user, I want the installer to provide suggestions for causes
+    of failure. See the existing work for translating terraform error
+    messages that is being done in https://github.com/openshift/installer/pull/3535.
+
+#### Extract relevant logs from the bootstrap
+
+  - As a user, I would like the installer to extract and display error
+    messages from bootstrap journal when relevant errors can be
+    identified.
+
+#### Implement bootstrap gather
+
+  - As a user, I want the installer to automatically gather logs when
+    bootstrap fails on the baremetal IPI platform, like it does for other
+    platforms.
+
+See also:
+ - https://github.com/openshift/installer/issues/2009
+
+#### Show errors from machine controllers
+
+  - As a user, I want the installer logs to bubble information up from
+    either machine-api-operator or cluster-baremetal-operator about why
+    workers failed to deploy. These operators should be degraded when
+    machine provisioning fails.
+
+#### Callback to Metal3
+
+  - As a user, I want my host to callback to Metal3/Ironic from ignition
+    when RHCOS boots.
+
+### Implementation Details/Notes/Constraints
+
+
+
+### Risks and Mitigations
+
+Some stories may impact the design of software managed by teams other
+than the baremetal IPI team. These including the installer and
+machine-api-operator teams, for example.
+
+## Design Details
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+### Upgrade / Downgrade Strategy
+
+Upgrades/downgrades are not applicable, as these are day 1
+considerations only. There is no impact on upgrades or downgrades.
```

Comment:
Improvements to having BareMetalHost status reflected in Operator status affects upgrades.  We may now be showing a Degraded `machine-api-operator` when we didn't before, blocking upgrades.  This sounds like what we'd want, but good to call out here.